### PR TITLE
feat(perf): Finish remaining landing v3 views 

### PR DESCRIPTION
**Pull request by k-fish**
### Summary

This adds a line chart type widget and support for different widget types on each of the views, as well as including multiple small fixes for behaviours noted as bugs. There will be another follow up PR for remaining small issues.

#### Other
- Made some small ui tweaks like fixing flex boxes for the mini-list widgets.

### CODE REVIEW

File: static/app/views/performance/charts/chart.tsx [hunk 1]
```diff
@@ -24,6 +25,7 @@ type Props = {
   disableXAxis?: boolean;
   chartColors?: string[];
   loading: boolean;
+  isLineChart?: boolean;
```
Comment (Zylphrex) [inline static/app/views/performance/charts/chart.tsx:28]: Are line charts and area charts the only charts we'll support here? If there will be additional chart types, perhaps something like
```typescript
chartType?: 'area' | 'line' | ...
```
may be more suitable

File: static/app/views/performance/charts/chart.tsx [hunk 2]
```diff
@@ -193,16 +199,31 @@ function Chart({
...
+      {zoomRenderProps => {
+        if (isLineChart) {
+          return (
+            <LineChart
+              height={height}
+              {...zoomRenderProps}
+              series={series}
+              previousPeriod={previousData}
+              xAxis={disableXAxis ? {show: false} : undefined}
+              {...areaChartProps}
```
Comment (Zylphrex) [inline static/app/views/performance/charts/chart.tsx:211]: This is essentially a duplicate of the `AreaChart` below, any chance we can get a helper function that returns the component class like we have in discover?

File: static/app/views/performance/charts/chart.tsx [hunk 3]
```diff
@@ -176,6 +179,9 @@ function Chart({
   };
 
   if (loading) {
+    if (isLineChart) {
+      return <LineChart height={height} series={[]} {...areaChartProps} />;
+    }
```
Comment (Zylphrex) [inline static/app/views/performance/charts/chart.tsx:184]: Can be a follow-up, but I think we typically use the `TransparentLoadingMask` over a chart to indicate loading states.

File: static/app/views/performance/landing/utils.tsx [hunk 4]
```diff
@@ -57,7 +57,6 @@ export const LANDING_DISPLAYS = [
     field: LandingDisplayField.MOBILE,
     isShown: (organization: Organization) =>
       organization.features.includes('performance-mobile-vitals'),
-    badge: 'new' as const,
```
Comment (Zylphrex) [inline static/app/views/performance/landing/utils.tsx:60]: Ah didn't get around to removing this. Thanks! 🥳

File: static/app/views/performance/landing/views/frontendOtherView.tsx [hunk 5]
```diff
@@ -14,7 +14,7 @@ export function FrontendOtherView(props: BasePerformanceViewProps) {
     <PerformanceDisplayProvider
       value={{performanceType: PROJECT_PERFORMANCE_TYPE.FRONTEND_OTHER}}
     >
-      <div data-test-id="frontend-pageload-view">
```
Comment (Zylphrex) [inline static/app/views/performance/landing/views/frontendOtherView.tsx:17]: Is this no longer needed?

File: static/app/views/performance/landing/widgets/components/widgetHeader.tsx [hunk 6]
```diff
@@ -17,10 +17,12 @@ export function WidgetHeader<T extends WidgetDataConstraint>(
   return (
     <WidgetHeaderContainer>
       <TitleContainer>
-        <StyledHeaderTitleLegend data-test-id="performance-widget-title">
-          {title}
-          <QuestionTooltip position="top" size="sm" title={titleTooltip} />
-        </StyledHeaderTitleLegend>
+        <div>
+          <StyledHeaderTitleLegend data-test-id="performance-widget-title">
+            <div className="truncate">{title}</div>
```
Comment (Zylphrex) [inline static/app/views/performance/landing/widgets/components/widgetHeader.tsx:22]: Should we be using `className` still?

File: static/app/views/performance/landing/widgets/widgetDefinitions.tsx [hunk 7]
```diff
@@ -105,6 +116,13 @@ export const WIDGET_DEFINITIONS: ({
     dataType: GenericPerformanceWidgetDataType.area,
     chartColor: WIDGET_PALETTE[3],
   },
+  [PerformanceWidgetSetting.P75_DURATION_AREA]: {
+    title: t('p75 Duration'),
+    titleTooltip: getTermHelp(organization, PERFORMANCE_TERM.P50),
+    fields: ['p75(transaction.duration)'], // TODO(k-fish): Check
```
Comment (Zylphrex) [inline static/app/views/performance/landing/widgets/widgetDefinitions.tsx:122]: What is this check for?

File: static/app/views/performance/landing/widgets/widgetDefinitions.tsx [hunk 8]
```diff
@@ -105,6 +116,13 @@ export const WIDGET_DEFINITIONS: ({
     dataType: GenericPerformanceWidgetDataType.area,
     chartColor: WIDGET_PALETTE[3],
   },
+  [PerformanceWidgetSetting.P75_DURATION_AREA]: {
+    title: t('p75 Duration'),
+    titleTooltip: getTermHelp(organization, PERFORMANCE_TERM.P50),
```
Comment (Zylphrex) [inline static/app/views/performance/landing/widgets/widgetDefinitions.tsx:121]:
```suggestion
    titleTooltip: getTermHelp(organization, PERFORMANCE_TERM.P75),
```

File: static/app/views/performance/data.tsx [hunk 9]
```diff
@@ -361,6 +362,7 @@ export const PERFORMANCE_TERMS: Record<PERFORMANCE_TERM, TermFormatter> = {
   frozenFrames: () => t('The count of the number of frozen frames in the transaction.'),
   mostErrors: () => t('Transactions with the most associated errors.'),
   mostIssues: () => t('The most instances of an issue for a related transaction.'),
+  slowHTTPSpans: () => t('The transactions with the slowest spans of a certain type.'),
```
Comment (Zylphrex) [inline static/app/views/performance/data.tsx:365]: Should there be custom messages for each type?

File: static/app/views/performance/landing/widgets/widgets/lineChartListWidget.tsx [hunk 10]
```diff
@@ -56,98 +58,126 @@ export function LineChartListWidget(props: Props) {
...
-            provided.widgetData.list.data[selectedListIndex].transaction as string,
+  const slowList = [
+    PerformanceWidgetSetting.SLOW_HTTP_OPS,
+    PerformanceWidgetSetting.SLOW_DB_OPS,
+    PerformanceWidgetSetting.SLOW_BROWSER_OPS,
+    PerformanceWidgetSetting.SLOW_RESOURCE_OPS,
+    PerformanceWidgetSetting.MOST_SLOW_FRAMES,
+    PerformanceWidgetSetting.MOST_FROZEN_FRAMES,
+  ];
+  const isSlowestType = slowList.includes(props.chartSetting);
```
Comment (Zylphrex) [inline static/app/views/performance/landing/widgets/widgets/lineChartListWidget.tsx:69]: Nit: Can we declare `slowList` as a set outside of this function

File: static/app/views/performance/landing/widgets/widgets/lineChartListWidget.tsx [hunk 11]
```diff
@@ -178,25 +209,58 @@ export function LineChartListWidget(props: Props) {
...
+                };
+                const rightValue =
+                  valueMap[isSlowestType ? 'slowest' : props.chartSetting];
+
                 switch (props.chartSetting) {
                   case PerformanceWidgetSetting.MOST_RELATED_ISSUES:
                     return (
                       <Fragment>
-                        <GrowLink to={transactionTarget}>
+                        <GrowLink to={transactionTarget} className="truncate">
```
Comment (Zylphrex) [inline static/app/views/performance/landing/widgets/widgets/lineChartListWidget.tsx:255]: Again, should we be using `className` still?

File: static/app/views/performance/landing/widgets/widgets/trendsWidget.tsx [hunk 12]
```diff
@@ -86,7 +86,7 @@ export function TrendsWidget(props: Props) {
         transform: transformTrendsDiscover,
       },
     };
-  }, [_eventView, trendChangeType]);
+  }, [eventView.query, eventView.fields, trendChangeType]);
```
Comment (Zylphrex) [inline static/app/views/performance/landing/widgets/widgets/trendsWidget.tsx:89]: I may be wrong but since `eventView` is derived from the location each time, wouldn't this cause `useMemo` to re-fire on every location change regardless if the `fields` changed or not?

**Review by Zylphrex — APPROVED** [review]

File: static/app/views/performance/charts/chart.tsx (hunk 1 above)
Comment (k-fish) [reply static/app/views/performance/charts/chart.tsx:28]: This file might go away entirely, which is why I wasn't too worried about using a one off prop like this. Good catch though.

File: static/app/views/performance/charts/chart.tsx (hunk 2 above)
Comment (k-fish) [reply static/app/views/performance/charts/chart.tsx:211]: Ditto above.

File: static/app/views/performance/charts/chart.tsx (hunk 3 above)
Comment (k-fish) [reply static/app/views/performance/charts/chart.tsx:184]: The transparent loading mask should only appear on reload state, which I'm thinking of handling in performance widget under the hood, instead of in these charts.

File: static/app/views/performance/landing/views/frontendOtherView.tsx (hunk 5 above)
Comment (k-fish) [reply static/app/views/performance/landing/views/frontendOtherView.tsx:17]: wrong data-test-id for frontend-other, not using it in a test yet so I yoinked it.

File: static/app/views/performance/data.tsx (hunk 9 above)
Comment (k-fish) [reply static/app/views/performance/data.tsx:365]: Yep. Thanks for reminding me.

File: static/app/views/performance/landing/widgets/widgets/lineChartListWidget.tsx (hunk 10 above)
Comment (k-fish) [reply static/app/views/performance/landing/widgets/widgets/lineChartListWidget.tsx:69]: Yeah, I'm also considering splitting the file. I'm trying to think if I can simplify this memoization code or list visualizations to reduce the boilerplate between widgets first.

File: static/app/views/performance/landing/widgets/widgets/lineChartListWidget.tsx (hunk 11 above)
Comment (k-fish) [reply static/app/views/performance/landing/widgets/widgets/lineChartListWidget.tsx:255]: Yes because the truncation below is separate, it handles truncating past a certain line length, but for a flex container that grows we also need this class to avoid overflow, or at least this is the fix I'm currently using to make sure the mini-lists don't overflow outside of the widget.

File: static/app/views/performance/landing/widgets/widgets/trendsWidget.tsx (hunk 12 above)
Comment (k-fish) [reply static/app/views/performance/landing/widgets/widgets/trendsWidget.tsx:89]: I still have to write some tests to ensure memoization is working correctly, likely as part of the batch query handler PR, but I _think_ this should be fine for now. `eventView.query` is a string, which can be compared as a primitive, which means it shouldn't cause it to refire if eventView is recreated, which is what was happening before (sorting the table caused widgets to reload... not ideal)

File: static/app/views/performance/landing/widgets/widgets/lineChartListWidget.tsx [hunk 13]
```diff
@@ -56,98 +58,126 @@ export function LineChartListWidget(props: Props) {
...
+              'medium'
+            )}
+          />
+        );
+      },
+      transform: transformEventsRequestToArea,
+    };
+  }, [
+    props.eventView.query,
+    props.fields[0],
```
[the remark this answers is not in the document]
Comment (k-fish) [reply static/app/views/performance/landing/widgets/widgets/lineChartListWidget.tsx:173]: Yep, great call. Might switch this when I normalize how widget specializations get called so it's always provided as the first field from the widget definition for all specializations.

File: static/app/views/performance/landing/widgets/components/widgetHeader.tsx (hunk 6 above)
Comment (k-fish) [reply static/app/views/performance/landing/widgets/components/widgetHeader.tsx:22]: Ditto above.

File: static/app/views/performance/landing/widgets/widgetDefinitions.tsx (hunk 7 above)
Comment (k-fish) [reply static/app/views/performance/landing/widgets/widgetDefinitions.tsx:122]: This can be nuked.

File: static/app/views/performance/landing/widgets/widgetDefinitions.tsx [hunk 14]
```diff
@@ -140,20 +158,90 @@ export const WIDGET_DEFINITIONS: ({
...
     fields: [`count()`],
     dataType: GenericPerformanceWidgetDataType.line_list,
     chartColor: WIDGET_PALETTE[0],
   },
+  [PerformanceWidgetSetting.SLOW_HTTP_OPS]: {
+    title: t('Slow HTTP Ops'),
+    titleTooltip: getTermHelp(organization, PERFORMANCE_TERM.SLOW_HTTP_SPANS),
+    fields: [`p75(spans.http)`],
+    dataType: GenericPerformanceWidgetDataType.line_list,
+    chartColor: WIDGET_PALETTE[0],
```
[the remark this answers is not in the document]
Comment (k-fish) [reply static/app/views/performance/landing/widgets/widgetDefinitions.tsx:208]: Yeah this is the 2nd approach where I'm just following exactly the colors in the designs that Dora laid out, originally I had it so the colours were sequential on the page, but then there is a lack of visual continuity.

tldr; good catch, I had already been thinking about this though so I'm prob just going to leave it for now and we can figure out what looks good around EA.
